src/mlpack/core/data/types_impl.hpp

```diff
@@ -0,0 +1,85 @@
+/**
+ * @file core/data/types.hpp
+ * @author Gopi M. Tatiraju
+ * 
+ * This file contains utilitiy fucntions related to types of data.
+ * We have adapted all the standard types which are available in armadillo.
+ * 
+ * This file also contains functions to convery external file types to mlpack
+ * file types. In future if we the user of mlpack needs support of an external
+ * linear algebra library like armadillo, all fucntions related to handling the
+ * types goes here.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_CORE_DATA_TYPES_IMPL_HPP
+#define MLPACK_CORE_DATA_TYPES_IMPL_HPP
+
+#include "types.hpp"
+
+namespace mlpack
+{
+namespace data
+{
```

Comment:
```suggestion
namespace mlpack {
namespace data {
```
Just a quick fix to match the style elsewhere in the library.  I am sure I have missed some things going through (and there are some things I've seen and haven't taken the time to comment about), so it's probably worth going through the whole PR and checking its consistency with the style guide. :+1: